movies-manager/src/containers/main/main.tsx

```diff
@@ -0,0 +1,199 @@
+        currentGenre: moviesGenres[0],
+    });
+    const [ greatestId, setGreatestId ] = useState(initGreatestId);
+    const [ movieWithDetails, setMovieWithDetails ] = useState(null);
+
+    useEffect(
+        () => {
+            if (!props.movieToAdd) {
+                return;
+            }
+            const currentId: number = greatestId + 1;
+            props.movieToAdd.id = currentId;
+            setMovies([ ...movies, props.movieToAdd]);
+            setGreatestId(currentId);
+        },
+        [props.movieToAdd]
+    );
+
+    const showModal: TShowModal = (modalType: string) => {
+        setIsFormDialogOpen(modalType === 'Edit');
+        setIsDeleteDialogOpen(modalType === 'Delete');
+    };
+
+    const hideModal: TVoidWithNoArgs = () => {
+        setIsFormDialogOpen(false);
+        setIsDeleteDialogOpen(false);
+    };
+
+    const handleMovieToEditChange: THandleMovie = (modalDialogType: string, id: number) => {
+        setMovieToEdit(movies.find((movie: IMovie) => movie.id === id));
+        showModal(modalDialogType);
+    };
+
+    const sortMoviesByField: TSortMoviesByField = (field: keyof IMovie) => {
+        const moviesCopy: IMovie[] = [ ...movies ];
+        if (field === 'release_date') {
+            moviesCopy.sort((movieA, movieB) => +new Date(movieA[field]) - +new Date(movieB[field]));
+        } else {
+            moviesCopy.sort((movieA, movieB) => (movieA[field] as number) - (movieB[field] as number));
+        }
+        setMovies( moviesCopy );
+    };
+
+    const updateMoviesSortConfig: TUpdateMoviesSortConfig = (isOpen: boolean, title?: TSortListItem) => {
+        const newSortConfig: Partial<ISelectConfig> = {
+            showOptionList: isOpen,
+            chosenOption: title,
+        };
+        setMoviesSortConfig({ ...moviesSortConfig, ...newSortConfig });
+        if (title) {
+            sortMoviesByField(title.replace(' ', '_') as keyof IMovie);
+        }
+    };
+
+    const updateMoviesSet: TUpdateMovieSet = (editableMovie: IMovie) => {
+        const movieIdx: number = movies
+            .findIndex(({ id }: IMovie) => id === movieToEdit.id);
+        const newMovies: IMovie[] = [ ...movies ];
+        newMovies.splice(movieIdx, 1, editableMovie);
+        setMovies( newMovies );
+        hideModal();
+    };
+
+    const setCurrentGenre: TSetGenre = useCallback(
+      (genre: TGenresListItem) => {
+        setMoviesGenresConfig({
+            ...moviesGenresConfig,
+            currentGenre: genre,
+        });
+      },
+      [moviesGenresConfig]
+    );
+
+    const deleteMovie: TVoidWithNoArgs = () => {
+        const movieIdx: number = movies
+            .findIndex(({ id }: IMovie) => id === movieToEdit.id);
+        const newMovies: IMovie[] = [ ...movies ];
+        newMovies.splice(movieIdx, 1);
+        setMovies( newMovies );
+        hideModal();
+    };
+
+    const showDetails: TShowDetails = (event: MouseEvent, movie: IMovie) => {
+        if (!event.ctrlKey) {
```

Comment:
Add ev.stopPropagation() in showEditMenu and hideEditMenu methods of movie-card to prevent use `event.ctrlKey`